lib/client-assets.php

```diff
@@ -807,13 +807,15 @@ function gutenberg_register_scripts_and_styles() {
 	);
 	wp_style_add_data( 'wp-components', 'rtl', 'replace' );
 
-	gutenberg_override_style(
-		'wp-block-library',
-		gutenberg_url( 'build/block-library/style.css' ),
-		current_theme_supports( 'wp-block-styles' ) ? array( 'wp-block-library-theme' ) : array(),
-		filemtime( gutenberg_dir_path() . 'build/block-library/style.css' )
-	);
-	wp_style_add_data( 'wp-block-library', 'rtl', 'replace' );
+	if ( gutenberg_post_has_blocks( get_the_ID() ) ) {
```

Comment:
Thanks for the PR.

The issue is that registering block styles is different than enqueuing block styles. If this style is being included in places where it shouldn't, we should fix the enqueuing issue instead.

Also, blocks are not loaded when we have a `get_the_ID()` result, they can be loaded in pages where we have multiple posts etc...

Reply:
Thanks for the answer.

I'm not sure that this is the correct solution for the problem, but it works.

I installed Wordpress, installed Gutenberg, there are no pages on the site. I've created an index.php, i go on site and what? -  I see the styles from the Gutenberg.

Why? The page does not contain any blocs that use the plugin and data styles.

I think it needs to be fixed, add styles only when the page is created in Gutenberg
Please see my screenshot
![screenshot_6](https://user-images.githubusercontent.com/22908236/48665940-a8b73700-eac0-11e8-84c6-28ab19fd7b3a.png)

Reply:
Loading these styles on the frontend is a conscious decision. There's no easy way to know that a page contains block or not especially since a page is not always about showing a specific page/post.

In Phase2 we plan to do some lazy-loading explorations to see if we can enqueue only the styles required by a given page. But even this is not a great approach because you'd end up with different stylesheets loaded for different posts which means the browser don't keep those in cache and will reload the styles on each page load.

The current option does add an extra stylesheet to load but at least this stylesheet don't change so it's only loaded once.

If you don't want this stylesheet to be loaded on your theme or installation, I suggest de-enqueuing it there https://codex.wordpress.org/Function_Reference/wp_dequeue_style